File: Prog54c/MainForm.rb
require "mscorlib"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

def int(text)     return text.to_i end
def float(text)   return text.to_f end
def str(text)     return text.to_s end
def list(obj)     return obj.to_a  end
def len(arr)      return arr.length end
def input(msg="") print msg; return gets end
def print(*args)  Kernel.print(*args, "\n") end
def round(x, y)   return float((x * 10**y).round) / 10**y end
def range(*args)  if len(args) == 1 then 
    return  list((0...args[0]).step(1)); elsif len(args) == 2 then 
    return  list((args[0]...args[1]).step(1)); elsif len(args) == 3 then 
    return  list((args[0]...args[1]).step(args[2])) end; end
class MyRandom;   def randint(min, max) return rand(max-min) + min; end; 
    def random() return rand() end; 
    def shuffle(arr) return arr.shuffle end; 
    def choice(arr) return arr[randint(0, len(arr))] end; 
end; $random = MyRandom.new(); $math = Math
MessageBox = System::Windows::Forms::MessageBox

class MainForm < System::Windows::Forms::Form
	def initialize()
		self.InitializeComponent()
	end

	def InitializeComponent()
		@button1 = System::Windows::Forms::Button.new()
		@button2 = System::Windows::Forms::Button.new()
		@button3 = System::Windows::Forms::Button.new()
		@textBox1 = System::Windows::Forms::TextBox.new()
		@label2 = System::Windows::Forms::Label.new()
		@label3 = System::Windows::Forms::Label.new()
		self.SuspendLayout()
		# 
		# button1
		# 
		@button1.BackColor = System::Drawing::Color.LightSlateGray
		@button1.Font = System::Drawing::Font.new("Tw Cen MT Condensed", 15.75, System::Drawing::FontStyle.Regular, System::Drawing::GraphicsUnit.Point, 0)
		@button1.ForeColor = System::Drawing::Color.Lavender
		@button1.Location = System::Drawing::Point.new(40, 340)
		@button1.Name = "button1"
		@button1.Size = System::Drawing::Size.new(160, 71)
		@button1.TabIndex = 0
		@button1.Text = "Calculate"
		@button1.UseVisualStyleBackColor = false
		@button1.Click { |sender, e| self.Button1Click(sender, e) }
		# 
		# button2
		# 
		@button2.BackColor = System::Drawing::Color.LightSlateGray
		@button2.Font = System::Drawing::Font.new("Tw Cen MT Condensed", 15.75, System::Drawing::FontStyle.Regular, System::Drawing::GraphicsUnit.Point, 0)
		@button2.ForeColor = System::Drawing::Color.Lavender
		@button2.Location = System::Drawing::Point.new(216, 340)
		@button2.Name = "button2"
		@button2.Size = System::Drawing::Size.new(160, 71)
		@button2.TabIndex = 1
		@button2.Text = "Clear"
		@button2.UseVisualStyleBackColor = false
		@button2.Click { |sender, e| self.Button2Click(sender, e) }
		# 
		# button3
		# 
		@button3.BackColor = System::Drawing::Color.LightSlateGray
		@button3.Font = System::Drawing::Font.new("Tw Cen MT Condensed", 15.75, System::Drawing::FontStyle.Regular, System::Drawing::GraphicsUnit.Point, 0)
		@button3.ForeColor = System::Drawing::Color.Lavender
		@button3.Location = System::Drawing::Point.new(394, 340)
		@button3.Name = "button3"
		@button3.Size = System::Drawing::Size.new(160, 71)
		@button3.TabIndex = 2
		@button3.Text = "Exit"
		@button3.UseVisualStyleBackColor = false
		@button3.Click { |sender, e| self.Button3Click(sender, e) }
		# 
		# textBox1
		# 
		@textBox1.Font = System::Drawing::Font.new("OCR A Extended", 8.25, System::Drawing::FontStyle.Regular, System::Drawing::GraphicsUnit.Point, 0)
		@textBox1.Location = System::Drawing::Point.new(216, 94)
		@textBox1.Name = "textBox1"
		@textBox1.Size = System::Drawing::Size.new(160, 19)
		@textBox1.TabIndex = 7
		# 
		# label2
		# 
		@label2.BackColor = System::Drawing::Color.LightSteelBlue
		@label2.Font = System::Drawing::Font.new("Tw Cen MT Condensed", 15.75, System::Drawing::FontStyle.Regular, System::Drawing::GraphicsUnit.Point, 0)
		@label2.ForeColor = System::Drawing::Color.MidnightBlue
		@label2.Location = System::Drawing::Point.new(40, 211)
		@label2.Name = "label2"
		@label2.Size = System::Drawing::Size.new(502, 42)
		@label2.TabIndex = 10
		# 
		# label3
		# 
		@label3.BackColor = System::Drawing::Color.LightSteelBlue
		@label3.Font = System::Drawing::Font.new("Tw Cen MT Condensed", 15.75, System::Drawing::FontStyle.Regular, System::Drawing::GraphicsUnit.Point, 0)
		@label3.ForeColor = System::Drawing::Color.MidnightBlue
		@label3.Location = System::Drawing::Point.new(40, 273)
		@label3.Name = "label3"
		@label3.Size = System::Drawing::Size.new(502, 42)
		@label3.TabIndex = 11
		# 
		# MainForm
		# 
		self.BackColor = System::Drawing::Color.Lavender
		self.ClientSize = System::Drawing::Size.new(584, 454)
		self.Controls.Add(@label3)
		self.Controls.Add(@label2)
		self.Controls.Add(@textBox1)
		self.Controls.Add(@button3)
		self.Controls.Add(@button2)
		self.Controls.Add(@button1)
		self.Name = "MainForm"
		self.Text = "Prog54c"
		self.Load { |sender, e| self.MainFormLoad(sender, e) }
		self.ResumeLayout(false)
		self.PerformLayout()
	end

	def Button3Click(sender, e)
		Application.Exit()
	end

	def Button2Click(sender, e)
		@textBox1.Text = ""
		@label2.Text = ""
		@label3.Text = ""
	end

	def Button1Click(sender, e)
		pi = float(3.14159)
		radius = float(@textBox1.Text)
		area = pi * radius**2
		area = round(area,3)
		circumference = 2 * pi * radius
		circumference = round(circumference,3)
		@label2.Text = "area: " + str(area)
		@label3.Text = "circumference: " + str(circumference)
		
		
	end

	def MainFormLoad(sender, e)
		
	end
end

File: prog82a/MainForm.py
﻿import System.Drawing
import System.Windows.Forms

from System.Drawing import *
from System.Windows.Forms import *

class MainForm(Form):
	def __init__(self):
		self.InitializeComponent()
	
	def InitializeComponent(self):
		self._textBox1 = System.Windows.Forms.TextBox()
		self._textBox2 = System.Windows.Forms.TextBox()
		self._label1 = System.Windows.Forms.Label()
		self._label2 = System.Windows.Forms.Label()
		self._label3 = System.Windows.Forms.Label()
		self._label4 = System.Windows.Forms.Label()
		self._button1 = System.Windows.Forms.Button()
		self._button2 = System.Windows.Forms.Button()
		self._button3 = System.Windows.Forms.Button()
		self.SuspendLayout()
		# 
		# textBox1
		# 
		self._textBox1.Location = System.Drawing.Point(267, 84)
		self._textBox1.Name = "textBox1"
		self._textBox1.Size = System.Drawing.Size(270, 20)
		self._textBox1.TabIndex = 0
		# 
		# textBox2
		# 
		self._textBox2.Location = System.Drawing.Point(267, 161)
		self._textBox2.Name = "textBox2"
		self._textBox2.Size = System.Drawing.Size(270, 20)
		self._textBox2.TabIndex = 1
		# 
		# label1
		# 
		self._label1.Font = System.Drawing.Font("OCR A Extended", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label1.ForeColor = System.Drawing.Color.SeaGreen
		self._label1.Location = System.Drawing.Point(70, 84)
		self._label1.Name = "label1"
		self._label1.Size = System.Drawing.Size(182, 38)
		self._label1.TabIndex = 2
		self._label1.Text = "Speed Limit:"
		# 
		# label2
		# 
		self._label2.Font = System.Drawing.Font("OCR A Extended", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label2.ForeColor = System.Drawing.Color.SeaGreen
		self._label2.Location = System.Drawing.Point(43, 156)
		self._label2.Name = "label2"
		self._label2.Size = System.Drawing.Size(200, 38)
		self._label2.TabIndex = 3
		self._label2.Text = "Vehicle Speed:"
		# 
		# label3
		# 
		self._label3.BackColor = System.Drawing.Color.DarkSeaGreen
		self._label3.Font = System.Drawing.Font("OCR A Extended", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label3.ForeColor = System.Drawing.Color.Honeydew
		self._label3.Location = System.Drawing.Point(267, 235)
		self._label3.Name = "label3"
		self._label3.Size = System.Drawing.Size(270, 38)
		self._label3.TabIndex = 4
		self._label3.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
		# 
		# label4
		# 
		self._label4.Font = System.Drawing.Font("OCR A Extended", 15.75, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._label4.ForeColor = System.Drawing.Color.SeaGreen
		self._label4.Location = System.Drawing.Point(156, 235)
		self._label4.Name = "label4"
		self._label4.Size = System.Drawing.Size(96, 38)
		self._label4.TabIndex = 5
		self._label4.Text = "Fine:"
		# 
		# button1
		# 
		self._button1.BackColor = System.Drawing.Color.SeaGreen
		self._button1.Font = System.Drawing.Font("OCR A Extended", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._button1.ForeColor = System.Drawing.Color.Honeydew
		self._button1.Location = System.Drawing.Point(34, 375)
		self._button1.Name = "button1"
		self._button1.Size = System.Drawing.Size(166, 75)
		self._button1.TabIndex = 6
		self._button1.Text = "Calculate"
		self._button1.UseVisualStyleBackColor = False
		self._button1.Click += self.Button1Click
		# 
		# button2
		# 
		self._button2.BackColor = System.Drawing.Color.SeaGreen
		self._button2.Font = System.Drawing.Font("OCR A Extended", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._button2.ForeColor = System.Drawing.Color.Honeydew
		self._button2.Location = System.Drawing.Point(220, 375)
		self._button2.Name = "button2"
		self._button2.Size = System.Drawing.Size(166, 75)
		self._button2.TabIndex = 7
		self._button2.Text = "Clear"
		self._button2.UseVisualStyleBackColor = False
		self._button2.Click += self.Button2Click
		# 
		# button3
		# 
		self._button3.BackColor = System.Drawing.Color.SeaGreen
		self._button3.Font = System.Drawing.Font("OCR A Extended", 18, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, 0)
		self._button3.ForeColor = System.Drawing.Color.Honeydew
		self._button3.Location = System.Drawing.Point(409, 375)
		self._button3.Name = "button3"
		self._button3.Size = System.Drawing.Size(166, 75)
		self._button3.TabIndex = 8
		self._button3.Text = "Exit"
		self._button3.UseVisualStyleBackColor = False
		self._button3.Click += self.Button3Click
		# 
		# MainForm
		# 
		self.BackColor = System.Drawing.Color.Honeydew
		self.ClientSize = System.Drawing.Size(599, 473)
		self.Controls.Add(self._button3)
		self.Controls.Add(self._button2)
		self.Controls.Add(self._button1)
		self.Controls.Add(self._label4)
		self.Controls.Add(self._label3)
		self.Controls.Add(self._label2)
		self.Controls.Add(self._label1)
		self.Controls.Add(self._textBox2)
		self.Controls.Add(self._textBox1)
		self.Name = "MainForm"
		self.Text = "prog82a"
		self.ResumeLayout(False)
		self.PerformLayout()


	def Button1Click(self, sender, e):
		limit = int(self._textBox1.Text)
		spd = int(self._textBox2.Text)
		numover = spd - limit
		fine = 5 * numover + 20
		self._label3.Text = str(fine)

	def Button2Click(self, sender, e):
		self._textBox1.Text = ""
		self._textBox2.Text = ""
		self._label3.Text = ""

	def Button3Click(self, sender, e):
		Application.Exit()